### ISSUE TITLE
hideDivider prop of SectionHeader is inverted

### ISSUE THREAD

**Issue by amcgee** [dhis2/ui#176]
The divider is shown iff `hideDivider` is specified, when it should be the opposite.

![Screen Shot 2020-06-17 at 5 06 39 PM](https://user-images.githubusercontent.com/947888/84915720-723b0700-b0bd-11ea-9c46-2ba286b80654.png)
![Screen Shot 2020-06-17 at 5 06 25 PM](https://user-images.githubusercontent.com/947888/84915723-736c3400-b0bd-11ea-95f6-28bbf7eb440e.png)

**Comment by HendrikThePendric** [comment]
I expect something else is going on @amcgee 
1. The logic in the `MenuSectionHeader` is correct, see:
https://github.com/dhis2/ui/blob/35b2964e13371998734f80e48ee8d0a3478f2ef9/packages/core/src/MenuSectionHeader/MenuSectionHeader.js#L26

2. But at the last moment we decided to add some magic to the `Menu`
https://github.com/dhis2/ui/blob/35b2964e13371998734f80e48ee8d0a3478f2ef9/packages/core/src/Menu/Menu.js#L23-L27

So the situation currently is as follows:
- By default, if the first `li` in a menu `ul` is a menu-section-header, the divider will be hidden, because it doesn't look good [*] to start the menu with a divider. 
- However, the current implementation does allow for that behaviour to be overridden by explicitly setting `hideDivider` to `false`.
[*] I think @ismay, @Mohammer5 and myself came to this conclusion during a meeting or a Slack conversation, because I can't seem to find a mention of this in the PR conversation here https://github.com/dhis2/ui/pull/44...

Given the confusion this "feature" has already caused, I wonder if we would perhaps want to get rid of it. But in any case, the above should at least explain what is causing the behaviour you're seeing.

**Comment by ismay** [comment]
It's not really clear what the intention of that bit of logic in the menu is. Is it:

* If hideDivider is not set, and it's the first item in the list, hide the divider
* If hideDivider is not set, and it's not the first item in the list, fall back to the dense prop

That second bit is confusing to me. Why is it falling back to the setting of the dense prop?

**Comment by Mohammer5** [comment]
The `dense` part doesn't make much sense to me either right now.
I think it should be

```js
hideDivider: index === 0
  // if it's the first item, hide the divider if hideDivider is not specified
  ? typeof props.hideDivider !== 'boolean' || props.hideDivider
  // if it's not the first item, show the divider if hideDivider is not specified
  : typeof props.hideDivider !== 'boolean' ? false : props.hideDivider
```

**Comment by HendrikThePendric** [comment]
Sorry @ismay and @Mohammer5, you guys are completely right. There's a bug in there too, see fix here #177.

(I mentioned this on Slack already, got to love all those separate communication channels)

**Comment by varl** [comment]
This looks like it is fixed.
